sympy/physics/optics/medium.py

```diff
@@ -152,6 +135,73 @@ def refractive_index(self):
         """
         return (c/self.speed)
 
+    def __lt__(self, other):
+        """
+        Compares based on refractive index of the medium.
+        """
+        return self.refractive_index < other.refractive_index
+
+    def __gt__(self, other):
+        return not self < other
+
+    def __eq__(self, other):
+        return self.refractive_index == other.refractive_index
```

Comment:
The `__eq__` method should be removed. Proper `Basic` subclasses shouldn't define `__eq__`.

Reply:
That will cause tests to fail. Should `_hashable_content` be set to just `refractive_index` then?

Reply:
No the tests should be changed.